recipes/notcurses/all/conanfile.py

```diff
@@ -0,0 +1,64 @@
+from conans import ConanFile, CMake, tools
+import os
+
+class NotcursesConan(ConanFile):
+    name = "notcurses"
+    description = "a blingful TUI/character graphics library"
+    topics = ("graphics", "curses", "tui", "console", "ncurses")
+    url = "https://github.com/conan-io/conan-center-index"
+    homepage = "https://nick-black.com/dankwiki/index.php/Notcurses"
+    license = "Apache-2.0"
+    exports_sources = ["CMakeLists.txt"]
+    settings = "os", "compiler", "build_type", "arch"
+    options = {"shared": [True, False]}
+    default_options = {"shared": False}
+    generators = "cmake", "pkg_config"
+
+    _cmake = None
+
+    def requirements(self):
+        self.requires("openimageio/2.2.7.0")
+        self.requires("libunistring/0.9.10")
+        self.requires("readline/8.0")
+        self.requires("ncurses.tinfo/6.2")
```

Comment:
No allowed

Reply:
how do i reference the tinfo component of ncurses? I don't want full ncurses -- I'll fail to link. I just need libtinfo from that distribution.

Reply:
requires ncurses, then select the proper component at link time of the underlying build files. Why does it fail to link with all ncurses?

Reply:
ahhh, it's not an issue with libncurses per se, it's that libtermcap.so is being dragged in alongside libtinfo.so, and they collide:

```
/usr/bin/ld: /home/dank/.conan/data/termcap/1.3.1/_/_/package/19729b9559f3ae196cad45cb2b97468ccb75dcd1/lib/libtermcap.a(termcap.c.o): in function `tputs':
termcap.c:(.text+0x2f0): multiple definition of `tputs'; /home/dank/.conan/data/ncurses/6.2/_/_/package/208e1714f3ce187f52ebc74d49beba71b8a3db65/lib/libtinfow.a(lib_tputs.o):lib_tputs.c:(.text+0x830): first defined here
/usr/bin/ld: /home/dank/.conan/data/termcap/1.3.1/_/_/package/19729b9559f3ae196cad45cb2b97468ccb75dcd1/lib/libtermcap.a(termcap.c.o):(.bss+0x8): multiple definition of `ospeed'; /home/dank/.conan/data/ncurses/6.2/_/_/package/208e1714f3ce187f52ebc74d49beba71b8a3db65/lib/libtinfow.a(lib_tputs.o):(.bss+0x4): first defined here
/usr/bin/ld: /home/dank/.conan/data/termcap/1.3.1/_/_/package/19729b9559f3ae196cad45cb2b97468ccb75dcd1/lib/libtermcap.a(termcap.c.o):(.bss+0x0): multiple definition of `PC'; /home/dank/.conan/data/ncurses/6.2/_/_/package/208e1714f3ce187f52ebc74d49beba71b8a3db65/lib/libtinfow.a(lib_tputs.o):(.bss+0x6): first defined here
collect2: error: ld returned 1 exit status
gmake[2]: *** [CMakeFiles/notcurses-core.dir/build.make:419: libnotcurses-core.so.2.1.7] Error 1
gmake[1]: *** [CMakeFiles/Makefile2:145: CMakeFiles/notcurses-core.dir/all] Error 2
gmake[1]: *** Waiting for unfinished jobs....
[ 47%] Linking CXX static library libnotcurses-core.a
[ 47%] Built target notcurses-core-static
gmake: *** [Makefile:149: all] Error 2
notcurses/2.1.7: 
notcurses/2.1.7: ERROR: Package 'db872feacefdb4864aec2649d2eb4e5f29bfe4da' build failed
notcurses/2.1.7: WARN: Build folder /home/dank/.conan/data/notcurses/2.1.7/_/_/build/db872feacefdb4864aec2649d2eb4e5f29bfe4da
ERROR: notcurses/2.1.7: Error in build() method, line 33
	cmake.build()
	ConanException: Error 2 while executing cmake --build '/home/dank/.conan/data/notcurses/2.1.7/_/_/build/db872feacefdb4864aec2649d2eb4e5f29bfe4da' '--' '-j64'
```

Reply:
yes you have an ODR violation between libs from `termcap` and `ncurses` recipes.

Actually, `termcap` comes from `readline` due to default option value `with_library=termcap`: https://github.com/conan-io/conan-center-index/blob/52858e5a147bd942b6a9af7cbd53a8b162a2974a/recipes/readline/all/conanfile.py#L23

As I said, you should be able to migate that by being very accurate on linked targets of your `target_link_libraries` calls in underlying build files.

Reply:
ok, thanks, i will look into this and try to get it resolved! thanks for all your patience and assistance. this will presumably require a new notcurses release (since it will require editing my build files), so i'll come back for 2.1.8.

Reply:
I was able to get past this by adding `"readline:with_library":"curses"` to `default_options`. Thanks for all the help!